### ISSUE TITLE
The string `coding:` on the first line of the file easily causes a coding issue.

### ISSUE THREAD

**Issue by HongchengFan** [python/mypy#16064]
**Description**

When a comment is placed on the first line of a file, a coding issue easily causes if the comment contains the string 'coding:' regardless of whether the developers intended to define encoding rules.

Here is a minimal code snippet to reproduce the crash:
```
# donotencoding:mypy
def f(x: int) -> int:
    return x + 1
```
Note: string `coding:` on the first line of the file.

Print the code in the file.py and execute mypy file.py.

**The Actual Output**

```
mypy: can't decode file 'file.py': unknown encoding: mypy
```

**The Expected Output**

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used:1.5.1
- Mypy command-line flags: mypy file.py --show-traceback
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.9.0
- Operating system and version: Ubuntu 20.04.6 LTS

**Comment by hauntsaninja** [comment]
This is intentional. The Python interpreter will also raise an error if you try to run this file.
